File: frontend/src/types/notes.ts
export interface Note {
  id: string;
  title: string;
  content: string;
  category: string;
  tags: string[];
  is_favorite: boolean;
  is_archived: boolean;
  created_at: string;
  updated_at: string;
}

export type NoteCategory = 
  | 'general'
  | 'bullet-points'
  | 'plans'
  | 'strategies'
  | 'wishlist'
  | 'other';

export interface CreateNoteRequest {
  title: string;
  content: string;
  category?: string;
  tags?: string[];
  is_favorite?: boolean;
}

export interface UpdateNoteRequest {
  title?: string;
  content?: string;
  category?: string;
  tags?: string[];
  is_favorite?: boolean;
  is_archived?: boolean;
}

export interface NotesResponse {
  notes: Note[];
  total: number;
  page: number;
  limit: number;
}

export interface SearchNotesParams {
  query?: string;
  category?: string;
  page?: number;
  limit?: number;
}

// API Error Response
export interface ApiError {
  error: string;
  details?: string;
}

// API Response wrapper
export interface ApiResponse<T> {
  data?: T;
  error?: ApiError;
  success: boolean;
}

File: backend/internal/dto/response/notes.go
package response

import (
	"time"

	"github.com/google/uuid"
)

// NoteResponse represents the response for note operations
type NoteResponse struct {
	ID         uuid.UUID `json:"id"`
	Title      string    `json:"title"`
	Content    string    `json:"content"`
	Category   string    `json:"category"`
	Tags       []string  `json:"tags"`
	IsFavorite bool      `json:"is_favorite"`
	IsArchived bool      `json:"is_archived"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

// NotesListResponse represents the response for listing notes
type NotesListResponse struct {
	Notes []NoteResponse `json:"notes"`
	Total int            `json:"total"`
	Page  int            `json:"page"`
	Limit int            `json:"limit"`
}
